get_hrc.py

```diff
@@ -81,20 +81,15 @@ def format_proton_data(dat, descrs):
         t[channel] = t[channel] * 1000.0
         tabs.append(t)
 
-    newdat = np.ndarray(len(tabs[0]), dtype=descrs)
-
-    newdat['satellite'] = tabs[0]['satellite']
-
-    time_ref = tabs[0]['time_tag']
-
+    # Get the union of times present in the per-channel tables
+    time_ref = set()
     for t in tabs:
-        if not all(t['time_tag'] == time_ref):
-            print(f'Warning: mismatch in channel time column')
-            sys.exit(0)
-        else:
-            for col in t.colnames:
-                if col in channels:
-                    newdat[col] = t[col]
+        time_ref = time_ref.union(set(t['time_tag']))
+    time_ref = np.array(sorted(list(time_ref)))
```

Comment:
Ah, this union stuff is crap.  I should have just used the unique times in `dat` instead of working with the tabs data structure.